src/game/common/bezfwditerator.cpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file
+ *
+ * @author xezon
+ *
+ * @brief Bezier Forward Iterator class.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "bezfwditerator.h"
+
+BezFwdIterator::BezFwdIterator()
+{
+    m_index = 0;
+    m_count = 0;
+    m_point.Zero();
+    m_add0.Zero();
+    m_add1.Zero();
+    m_add2.Zero();
+}
+
+BezFwdIterator::BezFwdIterator(int count, const BezierSegment *segment)
+{
+    m_point.Zero();
+    m_add0.Zero();
+    m_add1.Zero();
+    m_add2.Zero();
+    m_index = 0; // Thyme specific.
+    m_count = count;
+    m_segment = *segment;
+}
+
+void BezFwdIterator::Start()
+{
+    m_index = 0;
+
+    if (m_count > 1) {
+        const float distance = 1.0f / (m_count - 1);
+        const float sqr_distance = distance * distance;
+        const float cub_distance = distance * distance * distance;
+
+        D3DXVECTOR4 xvec(m_segment.m_points[0].x, m_segment.m_points[1].x, m_segment.m_points[2].x, m_segment.m_points[3].x);
```

Comment:
refer to my Evaluate_Bez_Segment_At_T comment for how to eliminate d3d

Reply:
Done.